static/src/javascripts/bootstraps/enhanced/football.js

```diff
@@ -0,0 +1,367 @@
+                    ).each(container => {
+                        matchDayFor(competition, resp.matchDate)
+                            .fetch(container)
+                            .then(
+                                () => {
+                                    extras[1] = {
+                                        name: 'Today’s matches',
+                                        importance: 2,
+                                        content: container,
+                                        ready: true,
+                                    };
+                                    renderExtras(extras, dropdownTemplate);
+                                },
+                                () => {
+                                    delete extras[1];
+                                    renderExtras(extras, dropdownTemplate);
+                                }
+                            );
+                    });
+                });
+            });
+        }
+    });
+
+    isCompetition((competition: string) => {
+        const $rightHandCol = $('.js-secondary-column').dim().height;
+        if ($rightHandCol === 0 || $rightHandCol > 1800) {
+            renderTable(competition, extras, dropdownTemplate);
+        }
+    });
+
+    isLiveClockwatch(() => {
+        const ml = new MatchListLive.MatchListLive(
+            'match-day',
+            isCompetition() || 'premierleague',
+            config.get('dateFromSlug')()
+        );
+        const $img = $('.media-primary');
+        const $matchListContainer = $.create(
+            '<div class="football-matches__container" data-link-name="football-matches-clockwatch"></div>'
+        ).css({
+            minHeight: $img[0] ? $img[0].offsetHeight : 0,
+        });
+
+        $img.addClass('u-h');
+        loading($matchListContainer[0], 'Fetching today’s matches…', {
+            text: 'Impatient?',
+            href: '/football/live',
+        });
+
+        $('.js-football-meta').append($matchListContainer);
+        ml
+            .fetch($matchListContainer[0])
+            .fail(() => {
+                ml.destroy();
+                $matchListContainer.remove();
+                $img.removeClass('u-h');
+            })
+            .always(() => {
+                if ($('.football-match', $matchListContainer[0]).length === 0) {
+                    ml.destroy();
+                    $matchListContainer.remove();
+                    $img.removeClass('u-h');
+                }
+                $matchListContainer.css({
+                    minHeight: 0,
+                });
+                loaded($matchListContainer[0]);
+            });
+    });
+
+    isFootballStatsPage(() => {
+        $('.js-chart').each(el => {
+            new TableDoughnut().render(el);
+        });
+    });
+
+    // Binding
+    bean.on(document.body, 'click', '.js-show-more-football-matches', e => {
```

Comment:
Is there more than one such element? If not, you could get rid of the `bean` dependency and event delegation.